test/unit_tests/event/preprocessor/ReceiptsMiddlewareSpec.ts

```diff
@@ -17,21 +17,29 @@
  *
  */
 
+import {ConversationRepository} from 'src/script/conversation/ConversationRepository';
 import {ClientEvent} from 'src/script/event/Client';
+import {EventService} from 'src/script/event/EventService';
 import {ReceiptsMiddleware} from 'src/script/event/preprocessor/ReceiptsMiddleware';
-import {noop, createRandomUuid} from 'Util/util';
+import {UserState} from 'src/script/user/UserState';
+import {noop} from 'Util/util';
+import {createRandomUuid} from 'Util/uuid';
 
 describe('ReceiptsMiddleware', () => {
   const selfId = createRandomUuid();
   let readReceiptMiddleware;
 
   beforeEach(async () => {
     const eventService = {loadEvents: noop, replaceEvent: noop};
-    readReceiptMiddleware = new ReceiptsMiddleware(eventService, undefined, {
-      self: () => ({
-        id: selfId,
-      }),
-    });
+    readReceiptMiddleware = new ReceiptsMiddleware(
+      eventService as unknown as EventService,
+      undefined as unknown as ConversationRepository,
+      {
+        self: () => ({
+          id: selfId,
+        }),
+      } as UserState,
+    );
```

Comment:
If it's converted to `ts` and fully functional, let's move it right next to the implementation file and rename it `.test.ts` (instead of `Spec`)